AutoHeuristic: tests

AlnisM opened the pull request:
This PR adds tests to AutoHeuristic that ensure that when existing heuristics are re-generated, the generated code stays the same.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133496



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

AlnisM committed:
AutoHeuristic: tests

[ghstack-poisoned]

AlnisM committed:
Update on "AutoHeuristic: tests"

This PR adds tests to AutoHeuristic that ensure that when existing heuristics are re-generated, the generated code stays the same.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

AlnisM committed:
Update on "AutoHeuristic: tests"

This PR adds tests to AutoHeuristic that ensure that when existing heuristics are re-generated, the generated code stays the same.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

eellison commented on torchgen/_autoheuristic/train_regression.py:
```diff
@@ -46,7 +46,7 @@ def main(self, log_path, other_datasets, nrows, heuristic_name):
 
         # We will do a grid search over the values
-        max_depths = [5, 10, 13, 15, 17, 20, 23, None]
+        max_depths = [5, 10]
```
why were these changed ? what does this entail ?

AlnisM replied:
We never want to use max_depths this large (the decision trees and the generated code will be huge), so I removed them, because trying out many different hyperparameters increases the time to learn the heuristic a lot. train_regression.py is only used for pad_mm, where a max_depth of 5 is the best choice, so it doesn't change anything for existing heuristics. I'll actually change it to max_depths = [5, 6, 7], because we probably dont want trees with max_depth > 7 and from my experience max_depth < 5 is not enough to learn a good heuristic.

eellison replied:
add this comment inline ?

eellison reviewed (approved):
great to have tests of regeneration, thank you!

AlnisM committed:
Update on "AutoHeuristic: tests"

This PR adds tests to AutoHeuristic that ensure that when existing heuristics are re-generated, the generated code stays the same.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

AlnisM commented:
@pytorchbot merge

AlnisM commented:
@pytorchbot merge

AlnisM commented:
@pytorchbot merge -i
